pyccel/codegen/printing/ccode.py

```diff
@@ -303,6 +303,28 @@ def _flatten_list(self, irregular_list):
             return [irregular_list]
 
     #========================== Numpy Elements ===============================#
+    def _largest_literal_subset(self, flattened_list):
+        """
+        Returns the longest subset of back to back literals in the given list starting from the first element
```

Comment:
This docstring doesn't seem to match the implementation. It seems to me that the function

Returns all consecutive scalars at the start of the list

Reply:
I did say `starting from the first element`, it is not clear?

Reply:
I was confused by the "longest subset". To me this sounds like you are going to look for all subsets of back to back literals and then return the longest. Without reading the code I would assume that "starting from the first element" means that no elements are excluded.

Also this is for scalars not literals

Reply:
Oh! I see

Reply:
https://github.com/pyccel/pyccel/pull/1128/commits/e7253e4d2c8753f330ef4732d9d027efb5161e70

Reply:
https://github.com/pyccel/pyccel/pull/1128/commits/1e19c5b59c527b5c8e51a25a98a6cb170114ed68: corrects wrong call

Reply:
@EmilyBourne